### ISSUE TITLE
Non-unique (per residue) atom names break ligand bonding structure in pdb files

### ISSUE THREAD

**Issue by iimry** [molstar/molstar#156]
Hey again,

I am loading in a PDB file that includes a protein and ligand. However, I see that the ligand is somehow bonded in a way that is not defined in the PDB file. After testing what is happening, I found that the ligand is not using unique atom names and somewhere in Molstar, it is creating new bonds because the ligand atom names are `N`, `C`, and `O`.

Is there a way to not have Molstar try to detect backbone atoms as it is still a valid PDB file (both VMD and NGL read the protein and ligand fine without creating bad bonds)?

Here's the ligand when the atom names are non-unique

<img width="590" alt="Screen Shot 2021-03-30 at 4 34 07 PM" src="https://user-images.githubusercontent.com/55925317/113007857-dc1d2c80-9176-11eb-89d1-1cb1a09173df.png">

And when the atom names are incremental (`N1`, `C2`, `C3`, `O4`, ...)

<img width="582" alt="Screen Shot 2021-03-30 at 4 33 48 PM" src="https://user-images.githubusercontent.com/55925317/113007943-eccda280-9176-11eb-9704-174151a5d653.png">

I'm attaching the two files that create these representations below (I used .txt because of GitHub, but they are .pdb)

[bad_ligand.txt](https://github.com/molstar/molstar/files/6229823/bad_ligand.txt)
[good_ligand.txt](https://github.com/molstar/molstar/files/6229824/good_ligand.txt)

Let me know if anything wasn't clear.
Thanks

**Comment by stefdoerr** [comment]
This sounds like quite a weird limitation from the mmCIF format. But I understand it's not up to you and not easy to change. 
I guess the best course of action then is to modify our apps to produce separate SDF for the ligand and PDB for the protein and load both in Mol*. 
Thanks!

**Comment by arose** [comment]
> This sounds like quite a weird limitation from the mmCIF format.

Yeah, the `atom_site.id` is not used at all in struct_conn records (https://mmcif.wwpdb.org/dictionaries/mmcif_pdbx_v40.dic/Images/Categories/struct_conn_neighbors_archive.svg).

**Comment by arose** [comment]
> I guess the best course of action then is to modify our apps to produce separate SDF for the ligand and PDB for the protein and load both in Mol*.

Probably, that way you can also provide bond-orders and be 100% sure that only the bonds are used that you define.

Nevertheless we might fix the issue at hand.

**Comment by papillot** [comment]
I am reopening the issue, as I think there is an undesirable side-effect with the fix, regarding altlocs: altlocs should have the same atom names, but now the atom name for altloc B is corrupted.
Here is an example using PDB file 6FUX. Ligand SRY has 1 atom altloc `OG2`, with alternates `A` and `B`.

<img width="530" height="405" alt="Image" src="https://github.com/user-attachments/assets/040dced7-79e7-4559-9852-a300b05d3c9c" />

Alternate B has name `OG2_1` instead of plain `OG2`.

Changing the previous code to ignore the label duplication for alternates, would preserve `_struct_conn` table integrity.
The current implementation adds some complexity to the bond perception feature I am working on, because it makes atom names unreliable for PDBs with altlocs. I would rather fix the issue in atom-site.ts, instead of patching it. Is that OK?

**Comment by arose** [comment]
> I would rather fix the issue in atom-site.ts, instead of patching it. Is that OK?

Yes, that sounds good. Just please make sure there is ample tests to help clarifying the intended behavior. Thank you.

**Comment by papillot** [comment]
Closed as resolved
